### DISCUSSION TITLE
[QUESTION] Admin level logic

### DISCUSSION THREAD

**Discussion by Portur** [wmgeolab/geoBoundaries#3197] [General]
Hi there,

Thanks for the effort with all of this. I've been scanning and downloading and viewing so much I forgot what I wanted in the first place!

I want to understand the logic behind assigning admin levels. I am familiar with admin levels from OSM and in the past has caused great pain in trying to synchronise what a local would define as a province. My logic thus far has been relying on admin level 4, but with many specific conditions for each country and each differing OSM database. The goal is to have the closest child to the country.

In the api example of `https://www.geoboundaries.org/api/current/gbOpen/FRA/ADM1/` I saw ADM1 refers to what I would see as a province or the closest child and delightfully so. Comparing admin level from OSM and geoBoundaries for Andorra as an example, It looks like you are using OSM equivalent admin level 7 as the ADM1.

What is the logic of admin level from geoBoundaries' side and is there any correlation to OSM admin level?

**Comment by DanRunfola** [comment]
I just moved this over to a discussion @Portur -

Right now, we try to use official definitions (i.e., the definitions the country itself uses) to define administrative levels.  This is very distinct from the OSM definitions in some cases, but does overlap in others, depending on the nature of how it was entered into OSM.  

Taking Andorra for a moment, we actually are using OSM as the underlying source right now, but we manually identified that in Andorra the highest level of ADM is the Parish (i.e., see https://en.wikipedia.org/wiki/Parishes_of_Andorra , though I don't have the notes in front of me so don't know what the source of that identification was).  So in that specific case we would have ignored the OSM classification, and instead gone with the countries hierarchy, which we establish through external research.

**Reply by Portur** [reply]
Thanks @DanRunfola. 

Does that mean ADM1 is the next level down, not as the closest child but rather the subdivisions looking at each country in isolation? Do all ADM1 polygons then have ISO-3166-2 codes?

**Reply by DanRunfola** [reply]
Correct - our ADM's would be country-specific, and based on the country hierarchy itself.

I would love it if all ADM1 polygons did have ISO codes, but practically they do not.  The issue as I see it is the landscape simply evolves too quickly for the ISO coding to work effectively.  We provide them when we can, but it's pretty sparse.

**Reply by Portur** [reply]
Thanks @DanRunfola 

I really appreciate your feedback
